envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyIngressRoutesFactory.kt

```diff
@@ -168,9 +168,13 @@ class EnvoyIngressRoutesFactory(
             }
         val builder = RouteConfiguration.newBuilder()
             .setName("ingress_secured_routes")
-        if (properties.routes.headersToRemove.isNotEmpty()) {
-            builder.addAllRequestHeadersToRemove(properties.routes.headersToRemove)
+
+        when {
+            properties.ingress.headersToRemove.isNotEmpty() -> {
+                builder.addAllRequestHeadersToRemove(properties.ingress.headersToRemove)
+            }
```

Comment:
Not sure if `when` is better then `if` when we have only one branch, but why not :)